### DISCUSSION TITLE
Check which event triggered the workflow

### DISCUSSION THREAD

**Discussion by joaocasarin** [community/community#26784] [Actions]
<p>Well, I’d like to know if it’s possible to know which event triggered the workflow. I mean, my yaml file is like this:</p>
<pre><code class="lang-auto">name: test

on:
  workflow_dispatch:
  pull_request:
    branches:
      - main

jobs:
  test:
    if: ${{ }}
# and so on
</code></pre>
<p>In the line where I created the <code>if</code>, I’d like to check what event triggered file/workflow/job, the <code>workflow_dispatch</code> OR the <code>pull_request</code>.</p>
<p>Thanks!</p>

**Comment by rcowsill** [comment, marked as answer]
<p>You can check the event type like this:</p>
<pre><code class="lang-yaml">    if: ${{ github.event_name == 'pull_request' }}
</code></pre>

**Comment by joaocasarin** [comment]
<p>that’s exactly what i was looking for, thanks!</p>
<p>Btw, where can I find what values each property of <code>github.</code> can be? I mean, if I use <code>${{ github. }}</code>, what can I use after the dot?</p>

**Comment by Simran-B** [comment]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/learn-github-actions/contexts#github-context">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#github-context" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/7819991/181099887-f8623908-6552-45fa-a88d-22d08b5515c8.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/learn-github-actions/contexts#github-context" target="_blank" rel="noopener nofollow ugc">Contexts - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
